### ISSUE TITLE
Zone low battery not reported as state

### ISSUE THREAD

**Issue by ptr727** [ufodone/envisalink_new#63]
May be related to #61 where the system low battery is not reported.

Envisalink sent me an alert email that a sensor battery is low (battery powered window sensor).  
I did not find the corresponding low battery state reported in the integration.

I would like to see that sensors/zones have low battery state, such that standard HA low battery blueprints can be used on both the system battery and zone.

<img width="763" alt="image" src="https://github.com/ufodone/envisalink_new/assets/2061579/78cc4724-46fe-4ca2-94dc-79f856b760f6">

<img width="471" alt="image" src="https://github.com/ufodone/envisalink_new/assets/2061579/c4a2d36a-15f3-47b8-9630-babb1d47594d">

Debug logging was on, I did not see any "low batt" reported for anything other than the system battery, that I will replace today.
[home-assistant.envisalink.lowbat.log](https://github.com/ufodone/envisalink_new/files/13256894/home-assistant.envisalink.lowbat.log)

**Comment by ufodone** [comment]
I don't see anything in their protocol documentation that would make a low battery state possible on a per-zone basis.  There is a general "Sensor/Zone Low Battery" indication but it does not indicate which zone.  I see these general updates in your log file but nothing else that indicates a low battery on a specific zone.  If you tell me the zone number that is low battery then I can have another look to see if I missed something.

It's possible that the notification you are getting directly from EyezOn may not be exposed through the interface this integration uses.

**Comment by ptr727** [comment]
From recent events:
<img width="914" alt="image" src="https://github.com/ufodone/envisalink_new/assets/2061579/016820b4-d6e7-4dff-9497-605261cf9567">

Guest Bathroom Window is Zone 25.

**Comment by ufodone** [comment]
I had another look and don't see anything being sent by the Envisalink that would indicate a low battery on a specific zone so I don't think there's anything I can do to expose this through the integration.

**Comment by ptr727** [comment]
Thx for looking. Is there an avenue to ask EVL to add better support for HA, i.e. parity with EyezOn?

**Comment by ufodone** [comment]
You could try contacting EyezOn either directly or through their forums to see if they could expose this information via their TPI (third party interface).  That is what this integration uses to interface with the EVL.

My recollection is that they haven't expressed any interest in directly supporting HA in the past so it may be more effective asking how to get access to this information rather than any request to support HA.

**Comment by rct** [comment]
Is this a DSC panel?  

I complained to EyezOn about some of the TPI gaps like this a number of years back. The TPI and Envisalink follow an ancient DSC serial interface. Wireless zones weren't really defined yet.  They were largely uninterested/unresponsive to the idea of extending the TPI.

If I remember correctly, if the low battery condition persists long enough, I think you'll get a general zone fault from the panel via the TPI.

**Comment by ptr727** [comment]
I asked [here](https://forum.eyezon.com/viewtopic.php?t=5389&sid=db775c295cecfe1c5cfc93c1387f1d08).

**Comment by rct** [comment]
Here is [the thread I tried to start on their forum in 2017](https://forum.eyezon.com/viewtopic.php?p=13562&sid=474d8ebe8bf3f8c92853f3cd65761a6a#p13562).

I thought there was more discussion, but that might have been in one of the longer more generic threads. 

It's interesting that "GrandWizard" said that info wasn't available from the panel, yet the RS-422 connected gateway did have that information. 

Looks like in that thread a week elapsed between when I first noticed the low battery via the Telguard/iControl gateway and when I got some notice via EyezOn.

**Comment by ptr727** [comment]
[Seems](https://forum.eyezon.com/viewtopic.php?p=15633&sid=832fb8bdb2302eb21c4324c6dc54eb2b#p15633) dev support could be provided if a dev account is created?

**Comment by ufodone** [comment]
I did email Envisacor on Nov 30th as suggested by K-Man but have received no response.

**Comment by ufodone** [comment]
A quick update: Envisacor got back to me recently and it's looking promising that they'll be able to add this functionality such that I could expose the sensor battery state through the integration.  I don't have a timeline yet but will keep you updated.

**Comment by ufodone** [comment]
Some good news: they have added new codes which I can use to detect low battery faults on the wireless sensors.  I haven't added the code yet but will try and get to it soon.  I don't have any wireless sensors myself so will need someone to confirm it for me.

A bit of unfortunate news though: they were not able to add it to the EVL3 firmware as they are no longer making anything but critical updates to those devices.  The feature is available on the EVL4, EVL4MAX and DUO units.

For the EVL4MAX and DUO devices, I'm told you can do this update yourself.  For the EVL4 units, EyezOn will need to push the update.  I'm told you will need to explicitly request the firmware update as they will not push it automatically.  The required firmware versions are:

```
EVL4 - 1.5.211
EVL4MAX - 1.0.44 aliased to 1.0.204
DUO - 1.0.95
```

I'll follow up when I have had a change to get the needed code changes in.

**Comment by ptr727** [comment]
Thank you for the update, unfortunate about EVL3, guess it is time to upgrade.

**Comment by ufodone** [comment]
I have created a [beta](https://github.com/ufodone/envisalink_new/releases/tag/v0.8.0-beta1) release with this functionality present.

**Comment by tinuva** [comment]
Requested a firmware update from eyezon, after it took me a few days to get my EVL4 released from a dealer so that I could add it to my own account. Because it seems you need to request it by making a support request after logging in on https://portal.eyezon.com/app/login.php and adding the EVL4 device to your account.

will let you know as soon as I received the upgrade and tested

**Comment by tinuva** [comment]
Ah I see now why one has to request it:

> if you upgrade to that version you will lose the local page for status
> and control, Radio Thermostat support and TPI login notifications.

**Comment by tinuva** [comment]
How do I test the low battery warnings, or rather what setup is required for this now that I am on the required firmware version?

**Comment by tinuva** [comment]
I think some feedback.

Even my non-wireless zones now show wireless battery sensors. I guess the integration can't yet figure out if a zone is wireless or not.

Perhaps a config entry to specify the wireless zones separate from the wired zones could fix this?

**Comment by ufodone** [comment]
> you will lose the local page for status

Hmm.. this concerns me as the integration uses the pages on port 80/8080 for discovery - most importantly to obtain the MAC address used as the unique ID for the integration.  Can you confirm whether there are any errors/warnings in your HA log file on startup?  The pages the integration queries are `http://<evl_address>/2` and `http://<evl_address>/3`.  

>I guess the integration can't yet figure out if a zone is wireless or not.

That's correct.

>Perhaps a config entry to specify the wireless zones separate from the wired zones could fix this?

I wasn't sure what to do about this.  My initial thought was that people could just hide/disable the entities they didn't care about in HA.  I inevitable have entities from other integrations I don't care about with no other way to get rid of them so it seemed consistent with that.  And having the sensors just show up seemed simplest for people configuring it.

But I could probably add a way to provide a list of the sensors that are wireless if that's peoples' preference.  I'd likely want to keep the main zone list as-is and have a separate list that allows tagging of the specific zones from that main list as wireless.

**Comment by tinuva** [comment]
Happy with your idea of the tagging of wireless zones, but up to you, if I have to hide the entities then thats ok. But I think the HA default is for integrations to not automatically enable all entities, so if you rather have them show up as disabled by default and the user enable them as they need them, that is better.

As for the errors, nothing that I seen, the only log entry that show up is:
```
2024-01-16 15:38:03.244 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration envisalink_new which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
```

That said. the biggest change is definitely on /2
![Screenshot 2024-01-16 at 18 16 30](https://github.com/ufodone/envisalink_new/assets/5417682/da7e4815-3c2f-41c5-a5b4-bf9945a87ce9)

For me /3 still looks fine:
![Screenshot 2024-01-16 at 18 16 40](https://github.com/ufodone/envisalink_new/assets/5417682/25010fae-992e-40b5-a053-d21374d7ccc9)

Integration seems to work fine. As far as I understand, the TPI is still there, just not the notifications. And really, I never used the zone list status or the ability to arm/disarm on /2

If you discovered the enabled zones on /2 that definitely won't work anymore, but I believe I have that configured in the zone list.

**Comment by rct** [comment]
> Ah I see now why one has to request it:
> 
> > if you upgrade to that version you will lose the local page for status
> > and control, Radio Thermostat support and TPI login notifications.

FWIW, I seem to recall some other EVL update broke TPI login notifications.

I find that unfortunate, since I use those as notifications that my EVL has rebooted and that I should check for new firmware (and lately watch for potential breakage)

**Comment by ufodone** [comment]
Yes, the removal of some features in the newer firmware versions is why you need to explicitly need to request an update.  According to the Envisacor developer I've been working with, they have had to remove some features to add new ones due to limited space on the devices for the firmware.  So unfortunately,  any new updates to EVL4 devices may be a compromise.  I don't think this applies to the newer hardware like the EVL4MAX though.

While I haven't tried it, I'd expect using an automation on the entities becoming unavailable should work.  The integration does change all entities to unavailable when the connection is lost.

**Comment by tinuva** [comment]
Yeah I get notifications when the module is unavailable.

```
Home Alarm
APP  [2:21 PM](**removed**)
Home alarm unavailable
[2:21](**removed**)
Home alarm Ready
[2:25](**removed**)
Home alarm unavailable
[2:26](**removed**)
Home alarm

**Comment by ufodone** [comment]
> I could probably add a way to provide a list of the sensors that are wireless

I've published a new beta release which does this.  Only zones in this list will have low battery entities created for them.

**Comment by tinuva** [comment]
Thank you, definitely helps :)

**Comment by ptr727** [comment]
I got my EVL4 so I can try this out, is there an easy way to upgrade configuration from an EVL3 to an EVL4?

**Comment by ufodone** [comment]
I tried simulating an upgrade and it seems to work.  If your new EVL4 is using the same IP address and password as the old one, then just restarting Home Assistant should probably be sufficient.  If the IP address or password has changed then you should just be able to go into the basic configuration options in the UI and update them to match.

**Comment by ptr727** [comment]
Ok great.  
What about upgrading the actual EVL device and configuration, is the zone info stored on the EVL or at EyezOn?

**Comment by ufodone** [comment]
From an HA perspective, I don't believe there's any configuration needed on the EVL device itself.  If you use the EyezOn portal and want to transfer configuration for that, then that's something you'll need to ask EyezOn about.  HA doesn't use any of that info as it's not available through the TPI.

**Comment by ufodone** [comment]
Released in [v0.8.0](https://github.com/ufodone/envisalink_new/releases/tag/v0.8.0).

**Comment by ufodone** [comment]
The discussion here is for the EVL supporting DSC panels.

I don't see anything in the Uno TPI documentation that indicates that it supports per-zone low battery notifications.  There is a general "Wireless Device in Low-Battery Fault" status on a partition but no indication of which zone it is on.  I don't have that wired up to an entity in HA at the moment.  There are a few other things that I haven't gotten around to adding for the Uno panels yet.
